packages/connect-form/src/hooks/use-connect-form.ts

```diff
@@ -129,27 +127,21 @@ export type UpdateConnectionFormField = (
 function buildStateFromConnectionInfo(
   initialConnectionInfo: ConnectionInfo
 ): ConnectFormState {
-  let connectionStringInvalidError = null;
-  let connectionStringUrl = new ConnectionStringUrl(defaultConnectionString);
-  try {
-    connectionStringUrl = new ConnectionStringUrl(
-      initialConnectionInfo.connectionOptions.connectionString
-    );
-  } catch (error) {
-    connectionStringInvalidError = (error as Error).message;
-  }
-
-  const connectionOptions = {
-    ...initialConnectionInfo.connectionOptions,
-  };
-
+  const [, errors] = parseConnectionString(
+    initialConnectionInfo.connectionOptions.connectionString
+  );
   return {
-    errors: [],
-    warnings: validateConnectionOptionsWarnings(connectionOptions),
-
-    connectionStringInvalidError,
-    connectionStringUrl,
-    connectionOptions,
+    errors: errors,
+    // Only enable connection string editing when it's the default string.
+    enableEditingConnectionString:
+      initialConnectionInfo.connectionOptions.connectionString ===
+      defaultConnectionString,
```

Comment:
And is not coming from a favorite or recent?

Reply:
Good call, updated.